gobblin-data-management/src/main/java/org/apache/gobblin/util/commit/CreateDirectoryWithPermissionsCommitStep.java

```diff
@@ -49,9 +48,16 @@ public class CreateDirectoryWithPermissionsCommitStep implements CommitStep {
   private final URI fsUri;
   private boolean isCompleted = false;
 
-  public CreateDirectoryWithPermissionsCommitStep(FileSystem targetFs, Map<String, List<OwnerAndPermission>> pathAndPermissions) {
+  public static final String CREATE_DIR_CONFIG_PREFIX = CreateDirectoryWithPermissionsCommitStep.class.getSimpleName();
+  public static final String STOP_ON_ERROR_KEY = CREATE_DIR_CONFIG_PREFIX + "stop.on.error";
+  public static final String DEFAULT_STOP_ON_ERROR = "true";
+
+  public final boolean stopOnError;
+
+  public CreateDirectoryWithPermissionsCommitStep(FileSystem targetFs, Map<String, List<OwnerAndPermission>> pathAndPermissions, Properties props) {
     this.pathAndPermissions = pathAndPermissions;
     this.fsUri = targetFs.getUri();
+    this.stopOnError = Boolean.parseBoolean(props.getProperty(STOP_ON_ERROR_KEY, DEFAULT_STOP_ON_ERROR));
```

Comment:
nit: it does stop, but more it seems a `failOnError` or `throwOnError` - emphasizing the fact that it will be fatal, more than that processing ceases